libvast/src/plugin.cpp

```diff
@@ -93,6 +93,10 @@ load(std::vector<std::string> bundled_plugins, caf::actor_system_config& cfg) {
   // Step 1: Get the necessary options.
   auto paths_or_names
     = caf::get_or(cfg, "vast.plugins", std::vector<std::string>{});
```

Comment:
I think this would still silently fail if we specify e.g. a number for `vast.plugins`. Probably a more robust approach would be like

```
if (caf::holds_alternative<std::vector<std::string>>(cfg, "vast.plugins))
  // ...
else if (caf::holds_alternative<std::string>(cfg, "vast.plugins"))
  // ...
else
  return make_error();
```

Also, i'm wondering what would happen if someone were to try to write something like this in the config, would this be regarded as a single string containing whitespace and comma?
```
vast:
  plugins: all, matcher
```

Reply:
> Also, i'm wondering what would happen if someone were to try to write something like this in the config, would this be regarded as a single string containing whitespace and comma?
> 
> ```
> vast:
>   plugins: all, matcher
> ```

Per the YMAL specification that would try to load a single plugin names "all, matcher", which would fail to load because it does not exist.

> I think this would still silently fail if we specify e.g. a number for `vast.plugins`. Probably a more robust approach would be like
> 
> ```
> if (caf::holds_alternative<std::vector<std::string>>(cfg, "vast.plugins))
>   // ...
> else if (caf::holds_alternative<std::string>(cfg, "vast.plugins"))
>   // ...
> else
>   return make_error();
> ```

Can't do that since it's still possible not to specify the key at all. Until we have some form of validation for our config I fear this cannot be implemented properly.

Reply:
> Can't do that since it's still possible not to specify the key at all

This could be worked around by a `put_missing` in front:
```
  caf::put_missing(opts, "vast.plugins", std::vector<std::string>{});
```